### PR TITLE
Support Rank Stabilized LoRA in the ModelConfig/LoraConfig

### PR DESCRIPTION
**Pull request by JohnGiorgi**
Tiny PR to support Rank Stabilized LoRA. To do this we:

- Expose the argument in `ModelConfig`. It is copied straight from https://huggingface.co/docs/peft/en/package_reference/lora#peft.LoraConfig.use_rslora
- Pass it through to `LoraConfig` in `utils.py`

Of course, a user could always do this outside TRL, but my motivation for enabling it this way was that I am working with a slightly modified version of the `sft.py` script, so it would be nice if I could enable Rank Stabilized LoRA with a single flag.

### CODE REVIEW

**Comment by kashif** [comment]
@JohnGiorgi  do we need to pin peft to some version for this?

**Review by kashif — APPROVED** [review]

**Comment by JohnGiorgi** [comment]
@kashif Good question! Looks like yes... [v0.8.0](https://github.com/huggingface/peft/releases/tag/v0.8.0) based on digging through the release notes

**Comment by kashif** [comment]
yes i believe we will need to pin peft to that for this PR, also could you kindly also add a test?

**Comment by JohnGiorgi** [comment]
@kashif Okay I bumped the peft dependency to >=0.8.0. I also added a test for `get_peft_config` (which looks to be untested), which also serves as a test to ensure `use_rslora` is getting passed from a `ModelConfig` to resulting `LoraConfig`

**Comment by kashif** [comment]
@JohnGiorgi i believe its failing for one of the older python with:
```python
AttributeError: 'str' object has no attribute 'removeprefix'
```

**Comment by JohnGiorgi** [comment]
Fixed!

File: trl/trainer/model_config.py [hunk 1]
```diff
@@ -64,6 +64,17 @@ class ModelConfig:
     lora_task_type: str = field(
         default="CAUSAL_LM", metadata={"help": "The task_type to pass for LoRA (use SEQ_CLS for reward modeling)"}
     )
+    use_rslora: bool = field(
+        default=False,
+        metadata={
+            "help": (
+                "When set to True, uses <a href='https://doi.org/10.48550/arXiv.2312.03732'>Rank-Stabilized LoRA</a>"
+                " which sets the adapter scaling factor to `lora_alpha/math.sqrt(r)`, since it"
+                " was proven to work better. Otherwise, it will use the original default"
```
Comment (qgallouedec) [inline trl/trainer/model_config.py:73]: > was proven to work better

do you have a ref?

Comment (qgallouedec) [reply trl/trainer/model_config.py:73]: Ah ok I get it, let me rephrase a bit the documentation.

File: setup.py [hunk 2]
```diff
@@ -71,15 +71,15 @@
 EXTRAS = {
     "test": [
         "parameterized",
+        "peft>=0.8.0"
         "pytest",
         "pytest-xdist",
-        "accelerate",
```
Comment (qgallouedec) [inline setup.py:76]: already in required

File: trl/trainer/model_config.py (hunk 1 above)
Comment (JohnGiorgi) [reply trl/trainer/model_config.py]: This was copied from https://huggingface.co/docs/peft/main/en/conceptual_guides/lora verbatim

**Comment by qgallouedec** [comment]
Merged, thank you @JohnGiorgi

**Comment by JohnGiorgi** [comment]
> Merged, thank you @JohnGiorgi

Thank you!
